### DISCUSSION TITLE
Signature of the on_response closure

### DISCUSSION THREAD

**Discussion by zrll12** [tower-rs/tower-http#431] [Q&A] [answered]
The documents about trace layer gives the following example:
```rust
.on_response(|response: &Response<Body>, latency: Duration, _span: &Span| {
    tracing::debug!("response generated in {:?}", latency)
})
```
This will generate a error
```
expected closure signature `for<'a, 'b> fn(&'a Response<http_body::combinators::box_body::UnsyncBoxBody<axum::body::Bytes, axum::Error>>, Duration, &'b Span) -> _`
   found closure signature `for<'a, 'b> fn(&'a Response<Body>, Duration, &'b Span) -> _`
```
And could run if change to 
```rust
.on_response(|response: &Response<http_body::combinators::UnsyncBoxBody<axum::body::Bytes, axum::Error>>, latency: Duration, _span: &Span| {
    //something
});
```
I wonder if there is a feature that make these two types compatible.

Also, is there a way to get the method and uri in the on_ response closure?

**Comment by davidpdrsn** [comment, marked as answer]
The type of the response body is dictated by the service, not the middleware. Looks like you’re using axum which uses that response body type.

You can just use the type that makes it compile. axum provides a type alias at `axum::body::BoxBody`
